pydantic/dataclasses.py

```diff
@@ -110,6 +110,15 @@ def _pydantic_post_init(self: 'Dataclass', *initvars: Any) -> None:
         if post_init_post_parse is not None:
             post_init_post_parse(self, *initvars)
 
+    def pydantic_dataclass_eq(self: 'Dataclass', other: Any) -> bool:
```

Comment:
This currently works **only** for automatically converted dataclasses.
This is to remove any regression due to the feature added in v1.7!
Maybe it could make sense to also support it natively like
```python
import dataclasses

import pydantic


@pydantic.dataclasses.dataclass
class PydanticDataclass:
    a: int
    b: float

@dataclasses.dataclass
class StdlibDataclass:
    a: int
    b: float


py_dc = PydanticDataclass(a=123, b=1.1)
std_dc = StdlibDataclass(a=123, b=1.1)
assert py_dc == std_dc
```

If we want to do this, this could be a new feature